yt_dlp/extractor/veo.py

```diff
@@ -0,0 +1,74 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+from ..utils import (
+    int_or_none,
+    mimetype2ext,
+    unified_timestamp,
+    url_or_none,
+)
+
+
+class VeoIE(InfoExtractor):
+    _VALID_URL = r'https?://app\.veo\.co/matches/(?P<id>[0-9A-Za-z-]+)'
```

Comment:
Veo URL can have underscores. The current regex does not catch them resulting in a 404 error when trying to fetch the video JSON.

Example of a faulty VEO URL: [https://app.veo.co/matches/20220313-2022-03-13_u15m-plsjq-vs-csl/](https://app.veo.co/matches/20220313-2022-03-13_u15m-plsjq-vs-csl/)

```suggestion
    _VALID_URL = r'https?://app\.veo\.co/matches/(?P<id>[0-9A-Za-z-_]+)'
```